app/services/service_catalog/add_to_order.rb

```diff
@@ -0,0 +1,21 @@
+module ServiceCatalog
+  class AddToOrder
+    attr_reader :order
+
+    def initialize(params)
+      @params = params
+    end
+
+    def process
```

Comment:
@mkanoor @syncrou What do you think about making a class method for `process` to simplify the caller?

The current controller call is:
```ruby
so = ServiceCatalog::AddToOrder.new(params)
render :json => so.process.order.to_hash
```

We could reduce this in two ways.
1. Create a call method so the caller does not have to call `new` on the class.
2. Return the `Order` object which is the only useful result

I was looking back at http://brewhouse.io/blog/2014/04/30/gourmet-service-objects.html and these two items where covered under `Real world services` and the `Values: The Return` section `Flavour #2: Return a persisted ActiveRecord model`

Makes the calling easier and removes the need for the `@order` instance variable and the `attr_reader`

The controller call would be reduced to:
```ruby
render :json => ServiceCatalog::AddToOrder.process(params)to_hash
```

Thoughts?

Reply:
@gmcculloug @syncrou 
I think the reason I chose returning an self and then plucking something from the response object was based on @Fryguy suggestion. With what you are suggesting the process would always return you back the result object (order in this example). @Fryguy wanted to return the service object and then the controller could pick what it needs to send back from the service object.

Reply:
I like @gmcculloug suggestion but if the consensus was to go the other way as suggested by @Fryguy, I'm fine with the current change.

Reply:
There were 2 reasons for why we did it this way.

1. In my mind, I was separating the presentation of the results from the actual ordering process.  So, I would expect the process method to do it's job and process.  Additionally, it would set accessors that other things could grab for presentation purposes.

2. the construct `ServiceCatalog::AddToOrder.process(params).to_hash` reads to me that we want to serialize the process itself, which I interpret as "did it run or did it blow up".  If you make process return an order the reader has no idea that an order is being rendered back to the user.  `ServiceCatalog::AddToOrder.new(params).process.order.to_hash` is more explicit to the reader as they can see that an order object is being serialized back to the user.

The chaining (i.e. returning `self`) was added for convenience.  I think I originally wanted it to return `true`, but then then caller was forced to code it as:

```ruby
service = ServiceCatalog::AddToOrder.new(params)
service.process
render :json => service.order.to_hash
```

Reply:
I could see a class method as a simpler entrypoint...

before

```ruby
ServiceCatalog::AddToOrder.new(params).process.order.to_hash
```

after

```ruby
ServiceCatalog::AddToOrder.process(params).order.to_hash
```

Doesn't really save very much, but ¯\\\_(ツ)_/¯